sdk/appconfiguration/azure-spring-cloud-feature-management/README.md

```diff
@@ -259,6 +259,104 @@ Options are available to customize how targeting evaluation is performed across
+
+The assignment-parameters work similarly to the other `FeatureFilters`. It can take in a `Map` which contains the evaluation criteria. In the case of `Microsoft.Targeting` it contains the `Audience` object. This enables custom assigners to be created.
+
+The `Microsoft.Targeting` assigner uses three evaluation criteria to assign a variant; users, groups, and defaultRolloutPercentage. Assignment requires the definition of the `ITargetingContextAccessor` where a user defines how the user and group parameters are defined.
+
+If a users is listed in a variant then they are automatically given that variant. If they are in more than one variant they are returned to the first listed variant.
+
+Groups have a rollout percentage. Using the user and group information users in groups are bucketed into parts of the rollout. This enables the a rollout where 10 of a group gets a variant, then when it increases to 50% that 10% still gets the variant with an additional 40% of the group now getting the variant. The rolloutPercentage for a group is shared across all variants that use that group.
+
+If Variant A has Group "Dev" which is set to 50, and Variant B also has Group "Dev" which is also set to 50 then, if you need to increase Variant A to have 60% for "Dev" then Variant B needs to be decreased to 40 as the max rollout percentage is 100.
+
+In a different setup Variant A has "Dev" set to 50 and Variant B has it set to 40. Then Variant A is increased to 60, then 10% of Variant B will move to use Variant A while the last 10% will be added to Variant B. This 10% of users could have had either variant before depending on the defaultRolloutPercentage configured and the default variant if one isn't assigned.
+
+A variant as shown can be part of multiple groups, as can a user be in multiple groups. Being in multiple groups results in multiple evaluation chances per group.
+
+If a user isn't listed, isn't in a group, or wan't in the rollout percentage of a group then they will be evaluated as part of the default rollout percentage. The default rollout percentage works the same as the group rollout percentage. It's value across all variants can't add up to more than 100% and each user is bucketed into a different part of the rollout. Because of this all users need to have a user id defined otherwise they will all return the same variant.
```

Comment:
wan't

Reply:
I think instead of "Because of this" it would be better to mention that the uniqueness of assignment in the group rollout comes from unique user id's up front rather than as a result of previous explanation.